src/main/java/com/xiaomi/infra/pegasus/client/PConfigUtil.java

```diff
@@ -61,13 +61,13 @@ public static Properties loadConfiguration(String configPath) throws PException
             new BufferedInputStream(
                 new FileInputStream(configPath.substring(PConfigUtil.LOCAL_FILE_PREFIX.length()))));
       } else if (PConfigUtil.isResource(configPath)) {
-        InputStream stream =
+        try (InputStream stream =
             PegasusClient.class.getResourceAsStream(
-                configPath.substring(PConfigUtil.RESOURCE_PREFIX.length()));
-        if (stream == null) {
+                configPath.substring(PConfigUtil.RESOURCE_PREFIX.length()))) {
+          config.load(stream);
+        } catch (NullPointerException e) {
```

Comment:
Thank you for your contribution, but I'm afraid this is not a cool way to fix the issue.
Issue #112  indicates that the stream was not closed after usage. I don't see wrapping around with a try-catch can fix this.

The following solution may be better:

```diff
        InputStream stream =
            PegasusClient.class.getResourceAsStream(
                configPath.substring(PConfigUtil.RESOURCE_PREFIX.length()));
        if (stream == null) {
          throw new PException("config resource not found: " + configPath);
        }
        config.load(stream);
+       stream.close();
```

Reply:
Ok, Fine, But I want to know why can't we use try-with-resource?

Reply:
You don't fix anything using try-with-resource AFAIS. There's no reason to change the code if nothing fixed.
What do you think of the benefit that adding try-catch here can bring?

Reply:
Thank you,😁 I understand that using it there is not a good way.